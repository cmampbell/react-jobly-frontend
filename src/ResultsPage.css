.ResultsPage div {
    margin-left: 10vh;
}