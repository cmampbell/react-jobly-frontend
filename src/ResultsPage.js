import React from 'react'
import { useLoaderData } from 'react-router-dom';
import './ResultsPage.css'

import SearchBar from './SearchBar'
import CompanyCard from './CompanyCard';
import JobCard from './JobCard';

// path should be either 'companies' or 'jobs'
const ResultsPage = ({path}) => {
    let results = useLoaderData();

    return (
        <div className="ResultsPage">
            <SearchBar path={path}/>
            {results.length > 0 ? results.map(result => path === 'companies' ?
                                        <CompanyCard company={result} key={result.handle}/>
                                        : <JobCard job ={result} key={result.id}/>)
                                : <p>No matching results.</p> }
        </div>
    )
}

export default ResultsPage;