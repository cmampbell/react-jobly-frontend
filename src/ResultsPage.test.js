import { render } from '@testing-library/react';
import ResultsPage from './ResultsPage';
import { RouterProvider, createMemoryRouter, Route } from 'react-router-dom';

const testCompanies = [
    {
    handle: 'test1',
    name: 'Name1',
    numEmployees: 45,
    description: 'Test company 1',
    jobs: [{id: 1, title: 'test1', companyName: 'test', salary: '50000', equity: '.03'},
            {id: 2, title: 'test2', companyName: 'test2', salary: '500', equity: '.05'}]
},
{
    handle: 'test2',
    name: 'Name2',
    numEmployees: 2,
    description: 'Test company 2'
}, 
{
    handle: 'test3',
    name: 'Name3',
    numEmployees: 101,
    description: 'Test company 3'
}, 
]

const route = [
    {
      path: "/",
      element: <ResultsPage path={'companies'} />,
      loader: ()=> testCompanies
    },
  ];


const router = createMemoryRouter(route)

describe('Smoke and snapshot CompanyPage test', () => {
  it('should render without crashing', () => {
    render(<RouterProvider router={router}>
      <ResultsPage path={'companies'}/>
    </RouterProvider>);
  });

  it('should match snapshot', async () => {
    const { asFragment, findByText } = render(<RouterProvider router={router}><ResultsPage path={'companies'}/></RouterProvider>)
    expect(asFragment()).toMatchSnapshot();
    expect(await findByText('Name1')).toBeInTheDocument()
  })
})
