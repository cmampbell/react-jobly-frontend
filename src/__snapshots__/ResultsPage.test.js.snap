// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot CompanyPage test should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ResultsPage"
  >
    <form
      action="/companies"
      method="get"
    >
      <input
        id="name"
        name="name"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search companies
      </button>
    </form>
    <a
      href="/companies/test1"
    >
      <div
        class="text-start card border-info"
        style="width: 80vw;"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Name1
          </h5>
          <p
            class="card-text"
          >
            Desc: Test company 1
          </p>
          <p
            class="card-text"
          >
            Size: 45 employees
          </p>
        </div>
      </div>
    </a>
    <a
      href="/companies/test2"
    >
      <div
        class="text-start card border-info"
        style="width: 80vw;"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Name2
          </h5>
          <p
            class="card-text"
          >
            Desc: Test company 2
          </p>
          <p
            class="card-text"
          >
            Size: 2 employees
          </p>
        </div>
      </div>
    </a>
    <a
      href="/companies/test3"
    >
      <div
        class="text-start card border-info"
        style="width: 80vw;"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Name3
          </h5>
          <p
            class="card-text"
          >
            Desc: Test company 3
          </p>
          <p
            class="card-text"
          >
            Size: 101 employees
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
